Create: Zero-shot translation

ktrnka opened the pull request:
# Description

Fixes #133 by adding an article on zero-shot translation

## Type of PR

- Creates the article _Zero-shot translation_

### Checklist:

- [x] I have read the [contributing guidelines](contributing.md).
- [x] I have followed the [style guide](http://machinetranslate.org/style).

## Challenges -- Could use advice

- I did a fairly quick literature review on this one, so the list of approaches may need updating. 
- I think I saw mention in papers of training encoder and decoder on monolingual data and finding a way to align the latent representations. I tried to word the transfer learning bullet to cover that, but it's not really the best name.
- I would've loved to indicate the general success of each approach but didn't have time for a full lit review to do it. I got the impression that bridging is generally better quality and more reliable than MNMT but also that systems are evaluated in unrealistic conditions by just omitting some data from high-resource language pairs, when the appeal of it is more for low-resource languages or to simplify the MLops of supporting many language pairs.

bittlingmayer commented on applications/advanced-concepts/zero-shot-translation.md:
```diff
@@ -0,0 +1,18 @@
+
+## Approaches
+
+- Bridging: If a language pair `(source, target)` is desired, this approach translates twice: `(source, bridge)`, then `(bridge, target)`. The `bridge` language is also called a pivot language. English is a common choice for the pivot because many languages have training data with English.
```
I might pull this out, because bridging has been going on since forever, whereas 'zero-shot' really only arose in connection with neural.

Rather, I'd frame this as an alternative or precursor to zero-shot.

bittlingmayer replied:
Also, we need a separate article on bridging anyway.  So maybe just make the article, using this text as a stub?

bittlingmayer commented on applications/advanced-concepts/zero-shot-translation.md:
```diff
@@ -0,0 +1,18 @@
+Zero-shot translation is desirable because it can be too costly to create training data for each language pair.
+
+Zero-shot machine translation is an active area of research.
+Zero-shot models are typically lower quality than supervised translation models.
```
One challenge here is that folks who don't know what zero-shot is probably don't know what supervision is.

bittlingmayer commented on applications/advanced-concepts/zero-shot-translation.md:
```diff
@@ -0,0 +1,18 @@
+- Bridging: If a language pair `(source, target)` is desired, this approach translates twice: `(source, bridge)`, then `(bridge, target)`. The `bridge` language is also called a pivot language. English is a common choice for the pivot because many languages have training data with English.
+- Multilingual [neural machine translation](/approaches/neural-machine-translation.md) (MNMT): This approach learns a single model for all language pairs. The target language is an input to the model. This approach can produce translations between languages that both have parallel data, but not necessarily parallel data with each other. 
+- Transfer learning: These approaches connect an encoder for the source language to a decoder for the target language. Algorithms are developed to align the latent representations of encoders and decoders across languages.
+
```
"Algorithms are developed" is a bit vague.

ktrnka replied:
From the quick and dirty lit review it seemed like there were two approaches:
1. Train encoder-decoders for a bunch of parallel texts (many pairs, like MNMT), make an effort to align latent representations, then stitch together the relevant encoder and decoder for the new language pair -- from <div class="csl-entry">Ji, B., Zhang, Z., Duan, X., Zhang, M., Chen, B., &#38; Luo, W. (2020). Cross-Lingual Pre-Training Based Transfer for Zero-Shot Neural  Machine Translation. <i>AAAI</i>. www.aaai.org</div>
2. Train denoising autoencoders on monolingual data and _somehow_ align the latent representations and do the same. I think these are typically called unsupervised MT. A couple refs that I haven't read yet -- https://aclanthology.org/2021.emnlp-main.533/  https://aclanthology.org/D18-1101/  Though also I the general idea

I grouped them together cause they seemed so similar but I didn't see a good naming convention for it and transfer learning seems too generic. I'll see what I can do at least about the wording

bittlingmayer commented:
Thanks for this contrib!

I think it's worth somehow tying it to broader concepts outside of machine translation:
- zero-shot learning
- transfer learning
We should mention these and read to some external resource, because, unlike for machine translation, there is plenty of reading on these concepts.

I think our policy should be that linking to Wikipedia is fine for such cases, since it's out of scope.

bittlingmayer commented:
Maybe one of us can add some concrete, evergreen example?

> In 2019, Goobooksoft launched a massively multilingual system that supported x languages and y language pairs.

I believe both Google and Facebook have launched this.

And I can say ModelFront has launched it for quality prediction.

bittlingmayer commented:
> I would've loved to indicate the general success of each approach but didn't have time for a full lit review to do it.

Minimal is fine and even good!  We also want to be evergreen.  If it isn't basically common knowledge for a person with your qualifications, then it's probably too detailed and/or unstable for the general audience.

ktrnka commented:
Ok, I _think_ I addressed all the comments except this.

> Maybe one of us can add some concrete, evergreen example?
> 
> > In 2019, Goobooksoft launched a massively multilingual system that supported x languages and y language pairs.
> 
> I believe both Google and Facebook have launched this.
> 
> And I can say ModelFront has launched it for quality prediction.

[This page](https://ai.googleblog.com/2022/05/24-new-languages-google-translate.html) indicates the use of MNMT in Google Translate from earlier this year.  I found a lot of publications by Google and Facebook on MNMT as well, but most don't say whether it was released or not.

I think we have a couple options:
- We can say "As of 2023, Google Translate uses MNMT to support ..." where we know about it
- We could link to relevant publications, though those may go out of date

cefoo commented on applications/advanced-concepts/bridging.md:
```diff
@@ -0,0 +1,16 @@
+description: Two-step translation through an intermediate language
+---
+
+**Bridging** uses an intermediate language to translate between source and target languages.
```
Thank you for your PR! :)

I'd make the intro sentence a little bit more like a definition, to keep consistency with the rest of the articles.

Something like:
---

**Bridging** is a machine translation method that uses an intermediate language to translate between source and target. languages.

---

cefoo replied:
I wonder if hyperlinking the `languages` section is always necessary.  The thing is that most (if not all) articles use the term `language`. Do you think it is a good idea to link it every time or perhaps only in some articles?

ktrnka replied:
Only some, for instance if there is significant information in the language-specific pages that we're trying to link to. For instance, if we mention language families that might make sense to link. Or if we're trying to offer a pointer to figure out the most common scripts for each language.

cefoo commented on applications/advanced-concepts/bridging.md:
```diff
@@ -0,0 +1,16 @@
+
+## Applications
+
+- Translation without parallel data
```
I would link the `parallel data` article here.
The link is: [https://machinetranslate.org/parallel-data](https://machinetranslate.org/parallel-data)

ktrnka commented:
Oh also I should mention #67 so it gets linked and we can close

cefoo commented:
> Oh also I should mention #67 so it gets linked and we can close

Yes, true! I'll close it.
